lib/multiplexer/multiplexer.go

```diff
@@ -356,12 +379,33 @@ func (m *Mux) detect(conn net.Conn) (*Conn, error) {
 				return nil, trace.Wrap(err)
 			}
 
+			if newProxyLine.isSigned() {
+				m.Debugf("Received signed PROXYv2 header for connection %s -> %s", conn.RemoteAddr(), conn.LocalAddr())
+			}
+
+			if newProxyLine.isSigned() && m.CertAuthorityGetter == nil {
+				return nil, trace.BadParameter("received signed PROXY line, but can't verify it")
+			}
+
 			// If TLVs are empty we know it can't be signed, so we don't try to verify to avoid unnecessary load
 			if m.CertAuthorityGetter != nil && m.LocalClusterName != "" && len(newProxyLine.TLVs) > 0 {
 				err = newProxyLine.VerifySignature(m.context, m.CertAuthorityGetter, m.LocalClusterName, m.Clock)
-				if err != nil && !errors.Is(err, ErrNonLocalCluster) {
-					return nil, trace.Wrap(err, "could not verify PROXY signature")
+				if errors.Is(err, ErrNoHostCA) {
+					m.WithFields(log.Fields{
+						"src_addr": conn.RemoteAddr(),
+						"dst_addr": conn.LocalAddr(),
+					}).Warnf("Could not verify PROXY signature for connection %s -> %s - could not get host CA", conn.RemoteAddr(), conn.LocalAddr())
+					continue
 				}
+				if err != nil {
+					return nil, trace.Wrap(err, "could not verify PROXY signature for connection %s -> %s", conn.RemoteAddr(), conn.LocalAddr())
+				}
+				m.WithFields(log.Fields{
```

Comment:
It seems reasonable to me to have a mentioning of signed header in debug logs if we've got one. But I guess we can remove one of the two mentions, probably this one is better to have.

Reply:
Other loggings removed, only this one left.

Reply:
It's only for those connection with signed header, so not on tunnelled connections. It was useful for me during developing/debugging, and I had a lot more logging around it, of course it's in a big part because I was focused on this area specifically, but with IP propagation becoming part of the core flow it can become more important. Plus, header replaces client's ip for the connection and this log line is the only place that links new address with the raw remote/local addresses of connection. I guess it can be changed to `Tracef` but I feel it's better to have it in `Debugf`

Reply:
Yes, but it's not just about amount, but also if it's useful to add to all other logs we already have in that case 😅 
But ok, since you both (+ Forrest) think it's too much, I changed it to be `Tracef()` 👍